detekt-rules-exceptions/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/ErrorUsageWithThrowable.kt

```
@@ -0,0 +1,80 @@
+package io.gitlab.arturbosch.detekt.rules.exceptions
+
+import io.gitlab.arturbosch.detekt.api.CodeSmell
+import io.gitlab.arturbosch.detekt.api.Config
+import io.gitlab.arturbosch.detekt.api.Entity
+import io.gitlab.arturbosch.detekt.api.RequiresTypeResolution
+import io.gitlab.arturbosch.detekt.api.Rule
+import io.gitlab.arturbosch.detekt.rules.fqNameOrNull
+import org.jetbrains.kotlin.builtins.StandardNames
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.calls.util.getResolvedCall
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameOrNull
+import org.jetbrains.kotlin.types.KotlinType
+import org.jetbrains.kotlin.types.typeUtil.supertypes
+
+/**
+ * This rule reports usage of `error` method with [Throwable] parameter, i.e, `error(throwable)`.
+ * The above will result in `throw IllegalStateException(throwable.toString())` which doesn't
+ * provide any info about `throwable.message` or `stackTrace`.
+ * Instead, use `error(throwable.message ?: "No error message provided")` to rethrow the throwable
+ * as `IllegalStateException` with `throwable.message`.
+ *
+ * <noncompliant>
+ * fun foo() {
+ *     try {
+ *         // ... some code
+ *     } catch(e: IOException) {
+ *         // some addition handling
+ *         error(e)
+ *     }
+ * }
+ * </noncompliant>
+ *
+ * <compliant>
+ * fun foo() {
+ *     try {
+ *         // ... some code
+ *     } catch(e: IOException) {
+ *         // some addition handling
+ *         error(e.message)
```

`throw IllegalStateException(e)` should be the way to fix this kind of issue or just `throw e`. Otherwise you are shallowing the exception.

---

Nice catch, I was not aware that using error(e.message) will still trigger the swallowed exception warning

---

Updated at https://github.com/detekt/detekt/pull/7419/commits/165c52eaa6c219b1df40c7ee2df58ef3ecd1611e